### PR TITLE
added QHAdam optimizer and fixed tests

### PR DESCRIPTION
**Pull request by morganmcg1**
Adding the QHAdam optimizer which has shown promise on the fastai forums when combined with Lookahead

### CODE REVIEW

**Comment by morganmcg1** [comment]
Not sure why the checks failed, I followed the steps here : https://forums.fast.ai/t/submitting-pr-to-fastai-v2/54146

And I ran `notebook2script` as below

`~/fastai_dev/dev$ python notebook2script.py --fname 12_optimizer.ipynb`

Yet I still get this error

```
!!! Diff between nbs and scripts
!!!Remember to run notebook2script.py or script2notebook.py
##[error]Bash exited with code '1'.
```

File: dev/12_optimizer.ipynb [hunk 1]
```diff
@@ -587,7 +587,7 @@
     "state = average_grad(state, p,  mom=0.9, dampening=True)\n",
```
Comment (sgugger) [inline dev/12_optimizer.ipynb]: <p>There is no need to import partial here, it's in the basics imports of fastai so already in the state.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/fastai/fastai_dev/pull/295/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='90'/>

**Comment by sgugger** [comment]
Thanks for the PR! I have one minor issue to fix before merging if you don't mind. Will look at why the diff produces something if the problem persists.

File: dev/12_optimizer.ipynb (hunk 1 above)
Comment (morganmcg1) [reply dev/12_optimizer.ipynb]: Oh yep, will remove now

**Comment by morganmcg1** [comment]
Comment addressed :)

**Comment by sgugger** [comment]
Thanks!

**Pull request merged**